Exercise_8/CustomControls/CircularProgressBar.cs

```diff
@@ -0,0 +1,123 @@
+{
+    public class CircularProgressBar : View
+    {
+        private int value;
+        private int strokeWeight;
+        private Color highlightColor;
+        private Color normalColor;
+        private const int MarginValue = 20;
+        private readonly Paint paint = new Paint();
+
+        public int Value
+        {
+            get => value;
+            set
+            {
+                this.value = value;
+                this.Invalidate();
+            }
+        }
+
+        public int StrokeWeight
+        {
+            get => strokeWeight;
+            set
+            {
+                this.strokeWeight = value;
+                this.Invalidate();
+            }
+        }
+
+        public Color HighLightColor
+        {
+            get => highlightColor;
+            set
+            {
+                this.highlightColor = value;
+                this.Invalidate();
+            }
+            
+        }
+
+        public Color NormalColor
+        {
+            get => normalColor;
+            set
+            {
+                this.normalColor = value;
+                this.Invalidate();
+            }
+
+        }
+
+        protected CircularProgressBar(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
+        {
+
+        }
+
+        public CircularProgressBar(Context context) : base(context, null)
+        {
+        }
+
+        public CircularProgressBar(Context context, IAttributeSet attrs) : this(context, attrs, 0)
+        {
+            TypedArray typeArray = context.ObtainStyledAttributes(attrs, Resource.Styleable.SetProgressProperties, 0, 0);
+            Value = typeArray.GetInteger(Resource.Styleable.SetProgressProperties_value, 0);
+            StrokeWeight = typeArray.GetInteger(Resource.Styleable.SetProgressProperties_strokeWeight, 20);
+            HighLightColor = typeArray.GetColor(Resource.Styleable.SetProgressProperties_highlightColor, Color.Green);
+            NormalColor = typeArray.GetColor(Resource.Styleable.SetProgressProperties_normalColor, Color.White);
+
+        }
+
+        public CircularProgressBar(Context context, IAttributeSet attrs, int defStyleAttr) : this(context, attrs, defStyleAttr, 0)
+        {
+        }
+
+        public CircularProgressBar(Context context, IAttributeSet attrs, int defStyleAttr, int defStyleRes) : base(context, attrs, defStyleAttr, defStyleRes)
+        {
+        }
+
+        public override void Draw(Canvas canvas)
+        {
+            var halfWidth = (float)canvas.Width / 2;
+            var halfHeight = (float)canvas.Height / 2;
+            base.Draw(canvas);
+
+            paint.Color = highlightColor;
+            paint.Flags = PaintFlags.AntiAlias;
+            paint.SetStyle(Paint.Style.Stroke);
+            paint.StrokeWidth = StrokeWeight;
+            canvas.DrawCircle(halfWidth, halfHeight, Math.Min(halfWidth, halfHeight) - MarginValue, paint);
+
+            paint.Color = normalColor;
```

Comment:
Extract this lines of code
```C#
paint.Color = normalColor;

var size = Math.Min(canvas.Width, canvas.Height) - 2 * MarginValue;
var x = (halfWidth > halfHeight ? halfWidth - halfHeight : 0) + MarginValue;
var y = (halfWidth > halfHeight ? 0 : halfHeight - halfWidth) + MarginValue;
canvas.DrawArc(x, y, x + size, y + size, 270, (float)(360 * (100 - value) / 100.0), false, paint);
```
To a new method. Let `normalColor` `value` as params and call.

Then we can do like this
```C#
drawCircle(normalColor, 100);
drawCircle(highlightColor, value);
```

Reply:
Done